rpc/client/client.go

```diff
@@ -341,7 +341,32 @@ func (c *Client) Block(ctx context.Context, height *int64) (*ctypes.ResultBlock,
 
 func (c *Client) BlockByHash(ctx context.Context, hash []byte) (*ctypes.ResultBlock, error) {
 	// needs block store
-	panic("BlockByHash - not implemented!")
+	var h [32]byte
+	if hash == nil {
+		// how should this be handled?
```

Comment:
In Tendermint there is no extra handling for this. It's just passed to store, and error about block not found is returned.

Reply:
OK great, I have taken a look at the Tendermint one now too. Updated the code based on this too